fix: comments on SOQL files only allowed at the top of the file

dobladez opened the pull request:
WI: @W-8611529@ @W-8801365@

Before:

https://user-images.githubusercontent.com/152839/106290678-7de0d580-6229-11eb-97bb-db61b3567964.mov

After:

https://user-images.githubusercontent.com/152839/106290726-8802d400-6229-11eb-864b-e5a4ed5d1d45.mov

sfsholden commented:
@dobladez Could you please expand on what you mean by BOF? I'm unclear why this change was needed. Please also include the work item in the description so future developers can get further context if needed.

dobladez commented:
> @dobladez Could you please expand on what you mean by BOF? I'm unclear why this change was needed. Please also include the work item in the description so future developers can get further context if needed.

Sorry, I meant "beginning of file". I've changed the description. Also added the WIs. Thanks

lcampos commented on grammars/soql.tmLanguage:
```diff
@@ -5070,6 +5070,15 @@
           </dict>
         </dict>
       </dict>
+      <key>soqlHeaderComment</key>
```
This should be added directly to `soql.tmLanguage.template.yml` since the `.tmLanguage` files are a result of compiling the `*.yml`.

dobladez replied:
The change is already on `src/soql.tmLanguage.template.yml`. Am I missing something?

lcampos replied:
My bad, I completely missed those changes

lcampos reviewed (changes requested):
Should address the comment on `soql.tmLanguage`
